core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionEndpointV2.kt

```diff
@@ -0,0 +1,95 @@
+import org.takes.rs.RsWithStatus
+
+class ConflictDetectionEndpointV2 : Take {
+    override fun act(req: Request?): Response {
+        return try {
+            val body = RqPrint(req).printBody()
+            val request =
+                conflictRequestAdapter.fromJson(body)
+                    ?: return RsWithStatus(RsText("missing request body"), 400)
+
+            val minStartTime = request.trainRequirements.values.minBy { it.startTime }.startTime
+            val trainRequirements = makeTrainRequirements(request.trainRequirements, minStartTime)
+            val conflicts = detectConflicts(trainRequirements)
+            val res = makeConflictDetectionResponse(conflicts, minStartTime)
+
+            RsJson(RsWithBody(conflictResponseAdapter.toJson(res)))
+        } catch (ex: Throwable) {
+            ExceptionHandler.handle(ex)
+        }
+    }
+}
+
+private fun makeTrainRequirements(
+    trainReqs: Map<Long, TrainRequirementsRequest>,
+    startTime: ZonedDateTime
+): List<TrainRequirements> {
+    val trainRequirements = mutableListOf<TrainRequirements>()
+    for ((id, trainReq) in trainReqs) {
+        val delta = TimeDelta(between(trainReq.startTime, startTime).toMillis())
+        val routingRequirements = mutableListOf<ResultTrain.RoutingRequirement>()
+        for (routingReq in trainReq.routingRequirements) {
+            routingRequirements.add(
+                ResultTrain.RoutingRequirement(
+                    routingReq.route,
+                    (routingReq.beginTime + delta).seconds,
+                    routingReq.zones.map {
+                        ResultTrain.RoutingZoneRequirement(
+                            it.zone,
+                            it.entryDetector,
+                            it.exitDetector,
+                            it.switches,
+                            (it.endTime + delta).seconds
+                        )
+                    }
+                )
+            )
+        }
+        val spacingRequirements = mutableListOf<ResultTrain.SpacingRequirement>()
+        for (spacingReq in trainReq.spacingRequirements) {
+            spacingRequirements.add(
+                ResultTrain.SpacingRequirement(
+                    spacingReq.zone,
+                    (spacingReq.beginTime + delta).seconds,
+                    (spacingReq.endTime + delta).seconds,
+                    true
+                )
+            )
+        }
+        trainRequirements.add(TrainRequirements(id, spacingRequirements, routingRequirements))
+    }
+    return trainRequirements
+}
+
+private fun makeConflictDetectionResponse(
+    conflicts: Collection<ConflictDetectionResult.Conflict>,
+    startTime: ZonedDateTime
+): ConflictDetectionResponse {
+    return ConflictDetectionResponse(
+        conflicts.map {
+            Conflict(
+                it.trainIds,
+                startTime.plusSeconds(it.startTime.toLong()),
+                startTime.plusSeconds(it.startTime.toLong()),
```

Comment:
We lose some precision here, the seconds are rounded down. We could instead do something like `startTime.plus(Duration.ofMillis((it.startTime * 1000).toLong()))`

Though that is quite ugly. 

By the way the second line should be changed to `endTime`